src/main/java/com/iexec/worker/chain/ContributionService.java

```diff
@@ -150,4 +144,48 @@ public ContributionAuthorization getContributionAuthorization(String chainTaskId
         return contributionAuthorizationService.getContributionAuthorization(chainTaskId);
     }
 
+    /*
+    * TODO See if it possible to remove useless fields from ContributionAuthorization:
+    * remove/merge, define Contribution/ContributionAuthorization responsibilities
+     */
```

Comment:
Agreed, I think ContributionAuthorization has become more of a WorkerpoolAuthorization.

Reply:
Changed it!

Reply:
Thank you!